R/train.R

```diff
@@ -31,6 +31,10 @@
 train = function(learner, task, subset, weights = NULL) {
   learner = checkLearner(learner)
   assertClass(task, classes = "Task")
+  if (hasExpression(learner)) {
+    dict = getTaskDictionary(task = task)
```

Comment:
The `dict` should somehow contain information about `subset`. The learner only sees the subset task, and probably expects the parameters to behave accordingly.